.github/workflows/default.yml

```diff
@@ -0,0 +1,75 @@
+name: Build Project
+
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+    name: '[${{ matrix.os }}] Build plugin'
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: set up Java
+        uses: actions/setup-java@v2
+        with:
+          distribution: 'adopt'
+          java-version: 16
+
+      - name: Cache .gradle
+        uses: burrunan/gradle-cache-action@v1
+        with:
+          gradle-distribution-sha-256-sum-warning: false
+
+      - name: Gradle Wrapper Validation
+        uses: gradle/wrapper-validation-action@v1
+
+      - name: ./gradlew assemble
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: assemble
+
+      - name: ./gradlew check
+        uses: eskatos/gradle-command-action@v1
+        with:
+          arguments: check
+
+      - name: Upload test results
+        if: ${{ always() }}
+        uses: actions/upload-artifact@v2
+        with:
+          name: test-results-${{ matrix.os }}
+          path: "${{ github.workspace }}/**/build/reports/tests"
+
+  integration-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ ubuntu-latest, windows-latest ]
+        gradle: [ 6.8.3, current, rc ]
+        java: [ 8, 11, 15 ]
```

Comment:
1. Minimum supported Gradle version is `6.8`, but I decided to set `6.8.3` as it contains number of performance fixes and I assumed if someone uses `6.8` they can easily migrate to `6.8.3`. 
2. I test on Java 15 here to keep the configuration simple. Java 16 is not supported by `current` gradle 7.0 nor by the `rc` 6.9-rc1 version. I could exclude them but I though Java 16 is not that different from java 15 so 🤷. 
I assumed that if Java 16 support becomes problematic then someone will improve the config